python/ray/tune/examples/memnn_example.py

```diff
@@ -0,0 +1,279 @@
+        question_encoded = question_encoder(question)
+
+        # compute a "match" between the first input vector sequence
+        # and the question vector sequence
+        # shape: `(samples, story_maxlen, query_maxlen)`
+        match = dot([input_encoded_m, question_encoded], axes=(2, 2))
+        match = Activation("softmax")(match)
+
+        # add the match matrix with the second input vector sequence
+        response = add(
+            [match, input_encoded_c])  # (samples, story_maxlen, query_maxlen)
+        response = Permute(
+            (2, 1))(response)  # (samples, query_maxlen, story_maxlen)
+
+        # concatenate the match matrix with the question vector sequence
+        answer = concatenate([response, question_encoded])
+
+        # the original paper uses a matrix multiplication for this reduction step.
+        # we choose to use a RNN instead.
+        answer = LSTM(32)(answer)  # (samples, 32)
+
+        # one regularization layer -- more would probably be needed.
+        answer = Dropout(self.config.get("dropout", 0.3))(answer)
+        answer = Dense(vocab_size)(answer)  # (samples, vocab_size)
+        # we output a probability distribution over the vocabulary
+        answer = Activation("softmax")(answer)
+
+        # build the final model
+        model = Model([input_sequence, question], answer)
+        return model
+
+    def _setup(self, config):
+        self.train_stories, self.test_stories = read_data()
+        model = self.build_model()
+        rmsprop = RMSprop(lr=self.config.get("lr", 1e-3), rho=self.config.get("rho", 0.9))
+        model.compile(
+            optimizer=rmsprop,
+            loss="sparse_categorical_crossentropy",
+            metrics=["accuracy"])
+        self.model = model
+
+    def _train(self):
+        # train
+        self.model.fit(
+            [self.inputs_train, self.queries_train],
+            self.answers_train,
+            batch_size=self.config.get("batch_size", 32),
+            epochs=self.config.get("epochs", 1),
+            validation_data=([self.inputs_test, self.queries_test], self.answers_test),
+            verbose=0)
+        _, accuracy = self.model.evaluate(
+            [self.inputs_train, self.queries_train],
+            self.answers_train,
+            verbose=0)
+        return {"mean_accuracy": accuracy}
+
+    def _save(self, checkpoint_dir):
+        file_path = checkpoint_dir + "/model"
+        self.model.save(file_path)
+        return file_path
+
+    def _restore(self, path):
+        # See https://stackoverflow.com/a/42763323
+        del self.model
+        self.model = load_model(path)
+
+
+if __name__ == "__main__":
+    import ray
+    from ray.tune import Trainable, run
+    from ray.tune.schedulers import PopulationBasedTraining
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--smoke-test", action="store_true", help="Finish quickly for testing")
```

Comment:
The smoke-test doesn't actually seem to do anything? You should limit `training_iteration` to like 20 if `smoke_test` is True.

Also, maybe set num_samples to like 4? So that this runs better on our CI.